source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -1183,6 +1184,22 @@ message TransformationPermutePhiOperands {
 
 }
 
+message TransformationPropagateInstructionUp {
+
+  // Propagates the first instruction in the block into the block's predecessors.
```

Comment:
Can you be explicit that this will duplicate the instruction as many times as needed, and that OpPhi will be used to get the value the original instruction would have generated?

Reply:
I notice from the transformation that it's not necessarily the very first instruction; e.g. you skip OpPhis and other instructions.

I think it would be nice to say that it is the first non-OpPhi instruction (and to change the code so that it doesn't skip over other instructions; we can use your new MoveInstructionDown transformation in the fuzzer pass to push other incompatible instructions out of the way.

Reply:
Fixed comments in c74f0e3. See https://github.com/KhronosGroup/SPIRV-Tools/pull/3478#discussion_r462343281 on the discussion of the application of `TransformationMoveInstructionDown`.